manifests/config.pp

```diff
@@ -21,12 +28,13 @@
     recurse => $nomad::purge_config_dir,
   }
   -> file { 'nomad config.json':
-    ensure  => file,
-    owner   => $nomad::user,
-    group   => $nomad::group,
-    path    => "${nomad::config_dir}/config.json",
-    mode    => $nomad::config_mode,
-    content => $_config,
+    ensure       => file,
+    owner        => $nomad::user,
+    group        => $nomad::group,
+    path         => "${nomad::config_dir}/config.json",
+    mode         => $nomad::config_mode,
+    validate_cmd => '/usr/local/bin/config_validate.rb %',
```

Comment:
due to my above comment, I think the validate_cmd should be configureable and have different defaults, depending on the OS. If it's an AIO installation, use the script. Otherwise just `nomad config validate`. Also we should raise an upstream issue so nomad improves their validation.

Reply:
Right now, I raised a feature request to Hashiscorp to ask to improve the validation: https://github.com/hashicorp/nomad/issues/16968
Though, there can be new new cases to cover, because I have experience only with the features that I am using. Adding new cases is straight forward, as well as removing them, when Hashicorp solves the issue from their side. 
But it makes sense: I can make the command customizable.. .and I'll check the above for the ruby interpreter.

Reply:
same as above: init.pp now contains a customizable parameter, which can be set to nomad_validator, ruby_validator, or a string ending with ` %` (space and percent)